draft-ietf-httpbis-priority.md

```diff
@@ -284,11 +284,44 @@ set to `5` and the incremental parameter set to `true`.
 priority = u=5, i
 ~~~
 
-## Defining New Parameters
+## Defining New Parameters {#register}
 
-When attempting to extend priorities, care must be taken to ensure any use of
-existing parameters leaves them either unchanged or modified in a way that is
-backwards compatible for peers that are unaware of the extended meaning.
+New Priority parameters can be defined by registering them in the HTTP Priority
+Parameters registry.
+
+Registration requests are reviewed and approved by a Designated Expert, as per
+{{!RFC8126}}, Section 4.5. A specification document is appreciated, but not
+required.
+
+The Expert(s) should consider the following factors when evaluating requests:
+
+* Community feedback
+* If the value is sufficiently well-defined
+* Generic parameters are preferred over vendor-specific, application-specific or
+  deployment-specific values. If a generic value cannot be agreed upon in the
+  community, the parameter's name should be correspondingly specific (e.g., with
+  a prefix that identifies the vendor, application or deployment).
+
+Registration requests should use the following template:
+
+* Name: \[a name for the Priority Parameter that matches key\]
+* Description: \[a description of the parameter semantics and value\]
+* Reference: \[to a specification defining this parameter\]
+
+See the registry at <https://iana.org/assignments/http-priority> for details on
+where to send registration requests.
+
+### Guidance for Defining New Parameters
+
+Since unknown parameters are ignored, there is no parameter extension
+negotiation mechanism, and the absence of signals for the mandatory urgency and
+incremental parameters has special meaning care needs to be taken when designing
+new extension parameters. A peer that does not implement an extension parameter
+is unaware of its meaning, resulting in the application of the design and
+constraints of defined parameters that it does support. Urgency and incremental
+are mandatory to implement parameters, so new parameters should not change their
+interpretation or modify them in a way that is not backwards compatible or
+fallback safe.
```

Comment:
I think that this paragraph calls out two design constraints:
* New parameters should not adversely interfere with existing ones.
* Nnew parameters cannot modify the meaning of existing ones.

It might be possible to derive the second point from the first point, though I agree that it's worth pointing out.

OTOH, I tend to think that
* We should better discuss about endpoints and intermediaries.
* Negotiation can be performed in a hop-by-hop manner, so I am hesitant to state that as the requirement. The problem in the design space is the multi-hop nature of HTTP, therefore we should call that out instead.
* This guidance will apply to any existing parameter rather than just "u" and "i."

So how about something like:
_When defining new parameters, care must be taken so that they do not adversely interfere with prioritization performed by existing endpoints or intermediaries that do not understand the newly defined parameter. Since unknown parameters are ignored, new parameters should not change the interpretation of or modify the predefined parameters in a way that is not backwards compatible or fallback safe._